src/browser/modules/Editor/EditorFrame.tsx

```diff
@@ -94,6 +116,80 @@ export function EditorFrame({ bus }: EditorFrameProps): JSX.Element {
 
   useEffect(() => bus && bus.take(EXPAND, toggleFullscreen))
   useEffect(() => bus && bus.take(CARDSIZE, toggleCardView))
+  useEffect(() => {
+    let isStillMounted = true
+    // when a saved Project Script or Local Cache Script is clicked
+    // not sure at this point which it could be
+    bus &&
+      bus.take(EDIT_CONTENT, () => {
+        if (isStillMounted) {
+          setActiveProjectFile(null)
+          setActiveProjectFileStatus(null)
+        }
+      })
+    // only when a Project Script is clicked
+    bus &&
+      bus.take(SELECT_PROJECT_FILE, projectFile => {
+        if (isStillMounted) {
+          setActiveProjectFile(projectFile)
+          setActiveProjectFileStatus(null)
+        }
+      })
+    // when a non-Project File action sets content in the editor
+    bus &&
+      bus.take(SET_CONTENT, () => {
+        if (isStillMounted) {
+          setActiveProjectFile(null)
+          setActiveProjectFileStatus(null)
+        }
+      })
+    // start of Project File edit (ProjectFileButton)
+    bus &&
+      bus.take(EDIT_PROJECT_FILE_START, () => {
+        if (isStillMounted) {
+          setActiveProjectFileStatus('saving edit...')
+        }
+      })
+    // successful completion of Project File edit (ProjectFilesButton)
+    bus &&
+      bus.take(EDIT_PROJECT_FILE_END, () => {
+        if (isStillMounted) {
+          setActiveProjectFileStatus(null)
+        }
+      })
+    // Edit error (ProjectFilesButton)
+    bus &&
+      bus.take(PROJECT_FILE_ERROR, () => {
+        if (isStillMounted) {
+          setActiveProjectFileStatus('error saving...')
+        }
+      })
+
+    return () => {
+      isStillMounted = false
+    }
+  }, [])
+
+  useEffect(() => {
+    let isStillMounted = true
+    // clear editor if active Relate file is deleted from sidebar
+    bus &&
+      bus.take(REMOVE_PROJECT_FILE, (removedProjectFile: ActiveRelateFile) => {
+        if (isStillMounted) {
+          if (
+            activeProjectFile &&
+            removedProjectFile.directory === activeProjectFile.directory &&
+            removedProjectFile.name === activeProjectFile.name
+          ) {
+            bus.send(SET_CONTENT, { message: '' })
+          }
+        }
+      })
+
+    return () => {
+      isStillMounted = false
+    }
+  })
```

Comment:
missing second parameter?

Reply:
hmm, yes, didnt work in this case. Im going to try again.